ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/MajorQueryCompactor.java

```diff
@@ -72,10 +72,9 @@ public boolean run(CompactorContext context) throws IOException {
    * See {@link org.apache.hadoop.hive.conf.HiveConf.ConfVars#SPLIT_GROUPING_MODE} for the config description.
    */
   private List<String> getCreateQueries(String fullName, Table t, String tmpTableLocation) {
-    return Lists.newArrayList(new CompactionQueryBuilder(
+    return Lists.newArrayList(new CompactionQueryBuilderForMajor(
```

Comment:
You may create a factory class for returning the specific `CompactionQueryBuilder` implementations. Using this approach have some benefit:
- the decisive logic can be hidden in the factory method
- the implementation classes can be hidden from the compactors (the factory returns the base class), especially if you move the builders into a sub-package. better decoupling between compactors and query builders.
- the `getCreateQueries(), getCompactionQueries(), getDropQueries() `method can be moved to `QueryCompactor`